src/java/org/apache/cassandra/db/MutationVerbHandler.java

```diff
@@ -61,38 +61,29 @@ private void addSensorsToResponse(Message.Builder<NoPayload> response, Mutation
         // Add write bytes sensors to the response
         Function<String, String> requestParam = SensorsCustomParams::encodeTableInWriteBytesRequestParam;
         Function<String, String> tableParam = SensorsCustomParams::encodeTableInWriteBytesTableParam;
-        Collection<Sensor> sensors = RequestTracker.instance.get().getSensors(Type.WRITE_BYTES);
-        addSensorsToResponse(sensors, requestParam, tableParam, response);
+        Collection<Sensor> requestSensors = RequestTracker.instance.get().getSensors(Type.WRITE_BYTES);
+        addSensorsToResponse(requestSensors, requestParam, tableParam, response);
 
         // Add index write bytes sensors to the response
         requestParam = SensorsCustomParams::encodeTableInIndexWriteBytesRequestParam;
         tableParam = SensorsCustomParams::encodeTableInIndexWriteBytesTableParam;
-        sensors = RequestTracker.instance.get().getSensors(Type.INDEX_WRITE_BYTES);
-        addSensorsToResponse(sensors, requestParam, tableParam, response);
-
-        // Add internode message sensors to the response
-        for (PartitionUpdate update : mutation.getPartitionUpdates())
-        {
-            Context context = Context.from(update.metadata());
-            Optional<Sensor> internodeBytes = SensorsRegistry.instance.getSensor(context, Type.INTERNODE_MSG_BYTES);
-            internodeBytes.ifPresent(sensor -> {
-                byte[] bytes = SensorsCustomParams.sensorValueAsBytes(sensor.getValue());
-                String internodeBytesTableParam = SensorsCustomParams.encodeTableInInternodeBytesTableParam(context.getTable());
-                response.withCustomParam(internodeBytesTableParam, bytes);
-            });
-
-            Optional<Sensor> internodeCount = SensorsRegistry.instance.getSensor(context, Type.INTERNODE_MSG_COUNT);
-            internodeCount.ifPresent(sensor -> {
-                byte[] count = SensorsCustomParams.sensorValueAsBytes(sensor.getValue());
-                String internodeCountTableParam = SensorsCustomParams.encodeTableInInternodeCountTableParam(context.getTable());
-                response.withCustomParam(internodeCountTableParam, count);
-            });
-        }
+        requestSensors = RequestTracker.instance.get().getSensors(Type.INDEX_WRITE_BYTES);
+        addSensorsToResponse(requestSensors, requestParam, tableParam, response);
+
+        // Add internode bytes sensors to the response after updating each per-table sensor with the current response
+        // message size: this is missing the sensor values, but it's a good enough approximation
+        int perSensorSize = response.currentSize(MessagingService.current_version) / requestSensors.size();
+        requestSensors = RequestTracker.instance.get().getSensors(Type.INTERNODE_BYTES);
+        requestSensors.forEach(sensor -> RequestTracker.instance.get().incrementSensor(sensor.getContext(), sensor.getType(), perSensorSize));
+        RequestTracker.instance.get().syncAllSensors();
```

Comment:
I think the `syncAllSensors` is redundant (and in other handlers) - it is already taken care of when mutation is applied.

Reply:
But this comes _after_ the mutation is applied right?

Reply:
Oh right. In that case, the `syncAllSensors` call in the apply method (or when the row iterator is closed in the read case) is redundant. Should we just keep the call in the verb handler? Now that you made idempotent, I'm less concerned about it but still a small performance hit